java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/diff/ObfDiffMerger.java

```diff
@@ -118,24 +119,38 @@ private void mergeRelationDiff(String[] args) throws IOException {
 		BinaryMapIndexReader.MapIndex mi = commonObf.getMapIndex();
 		Integer rl = mi.getRule(OSMAND_CHANGE_TAG, OSMAND_CHANGE_VALUE);
 		int deleteId = rl == null ? -1 : rl;
+
+		HashSet<Long> deletedShiftedIds = new HashSet<>();
+		if (deleteId > 0) {
+			for (MapZooms.MapZoomPair mz : commonObf.getZooms()) {
```

Comment:
1. deleted ids could be better inside loop  ```for (MapZooms.MapZoomPair mz : commonObf.getZooms()) {```
2. The main question remains do we understand the reason why we have different ids